### ISSUE TITLE
Do not read/write DOM outside of $(document).ready

### ISSUE THREAD

**Issue by eliemichel** [Phyks/VelibFxos#52]
_Guidelines are meant to remain open issues, just for contributors information._

The Black Box pattern used in `Views.py` is a little bit weird.

For instance, `header` is built as soon as the file is loaded and its construction calls `$('#app-bar a')` which requires the DOM to be fully loaded.

Actually, using `foo = (function() { return 'bar' })()` seems strictly equivalent to `foo = bar` for me. This trick can be used when you need the whole file to be read before starting to execute it or to limit the scope of some variables but here it is not the goal.

You may read the [Black Box guideline](https://hacks.mozilla.org/2014/08/black-box-driven-development-in-javascript/) again.

**Comment by Phyks** [comment]
still an issue?

**Comment by Phyks** [comment]
And should use http://www.w3schools.com/tags/att_script_defer.asp.

**Comment by eliemichel** [comment]
Indeed, I was thinking about it too, although I don't know its support.

**Comment by Phyks** [comment]
According to last page, it is widely supported.
